packages/video_player/video_player_platform_interface/lib/video_player_platform_interface.dart

```diff
@@ -371,4 +377,41 @@ class VideoPlayerOptions {
   /// Note: This option will be silently ignored in the web platform (there is
   /// currently no way to implement this feature in this platform).
   final bool mixWithOthers;
+
+  /// Additional web controls
+  final VideoPlayerWebOptions? webOptions;
+}
+
+/// [VideoPlayerWebOptions] can be optionally used to set additional web settings
+@immutable
+class VideoPlayerWebOptions {
+  /// [VideoPlayerWebOptions] can be optionally used to set additional web settings
+  const VideoPlayerWebOptions({
+    this.controlsEnabled = false,
+    this.allowDownload = true,
+    this.allowFullscreen = true,
+    this.allowPlaybackRate = true,
+    this.allowContextMenu = true,
```

Comment:
Should we allow people set (most) of the Attributes listed here, that are still not available through the normal API?

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/video#attributes

I know people have asked for (or hacked the generated code to set): `muted`, `autoplay`, `poster`...